pkg/types/entries.go

```diff
@@ -36,6 +39,7 @@ type EntryImpl interface {
 	Canonicalize(ctx context.Context) ([]byte, error) // marshal the canonical entry to be put into the tlog
 	Unmarshal(e models.ProposedEntry) error           // unmarshal the abstract entry into the specific struct for this versioned type
 	CreateFromArtifactProperties(context.Context, ArtifactProperties) (models.ProposedEntry, error)
+	Verifier() (*x509.Certificate, crypto.PublicKey, openpgp.EntityList, error)
```

Comment:
I would recommend sending back a single type for this, and it would simplify what the interface for `PublicKey` would be. Could you return canonicalized PEM bytes for each of these types? 

Then verifiers/monitors could grab those bytes and unmarshal it themselves, which is a downside, but it makes this interface a LOT nicer

Reply:
I guess the thing was that you would have already had the canonicalized PEM bytes.

Instead, return a `signature.Verifier` then, because that'll directly be useful for you and be a single comomn interface for verification and a return time here.

Reply:
Updated based on conversation